hathor/cli/side_dag.py

```diff
@@ -0,0 +1,220 @@
+        if not is_option(arg):
+            continue
+
+        try:
+            value = None if is_option(argv[i + 1]) else argv[i + 1]
+        except IndexError:
+            value = None
+
+        argv_list = hathor_node_argv
+        if arg.startswith('--side-dag'):
+            arg = arg.replace('--side-dag-', '--')
+            argv_list = side_dag_argv
+
+        argv_list.append(arg)
+        if value is not None:
+            argv_list.append(value)
+
+    return hathor_node_argv, side_dag_argv
+
+
+def _run_side_dag_node(argv: list[str], *, hathor_node_process: Process, conn: 'Connection') -> None:
+    """Function to be called by the main process to run the side-dag full node."""
+    logger.info('waiting for hathor node to initialize...')
+    if not conn.poll(HATHOR_NODE_INIT_TIMEOUT):
+        logger.critical(
+            f'side-dag node not started because hathor node failed to initialize before {HATHOR_NODE_INIT_TIMEOUT} '
+            f'seconds timeout'
+        )
+        return
+
+    message = conn.recv()
+    if isinstance(message, HathorProcessInitFail):
+        logger.critical(f'side-dag node not started because hathor node initialization failed:\n{message.reason}')
+        return
+
+    assert isinstance(message, HathorProcessInitSuccess)
+    logger.info('hathor node initialized')
+    logger.info('starting side-dag node...')
+
+    try:
+        side_dag = SideDagRunNode(argv=argv)
+    except (BaseException, Exception):
+        logger.critical('terminating hathor node...')
+        conn.send(SideDagProcessTerminated())
+        hathor_node_process.terminate()
+        return
+
+    side_dag.run()
+
+    # If `run()` returns, either the hathor node exited and terminated us, leaving the message below, or the side-dag
+    # node exited and we will terminate the hathor node.
+    if conn.poll():
+        message = conn.recv()
+        assert isinstance(message, HathorProcessTerminated)
+        logger.critical('side-dag node terminated because hathor node exited')
+        return
+
+    conn.send(SideDagProcessTerminated())
+    logger.critical('terminating hathor node...')
+    hathor_node_process.terminate()
+
+
+def _start_hathor_node_process(argv: list[str], *, capture_stdout: bool, conn: 'Connection') -> Process:
+    """Create and start a Hathor node process."""
+    run_hathor_node_args = (argv, RunNode, capture_stdout, conn)
+    hathor_node_process = Process(target=_run_hathor_node, args=run_hathor_node_args)
+    hathor_node_process.start()
+    return hathor_node_process
+
+
+def _run_hathor_node(argv: list[str], run_node_cmd: type[RunNode], capture_stdout: bool, conn: 'Connection') -> None:
+    """Function to be called by a background process to run the Hathor full node."""
+    from hathor.cli.util import process_logging_options, setup_logging
+
+    # We don't terminate via SIGINT directly, instead the side-dag process will terminate us.
+    signal.signal(signal.SIGINT, lambda _, __: None)
```

Comment:
Will this suppress the normal behavior of raising a `KeyboardInterrupt` when the process receives a `SIGINT`?

Reply:
Yes!